daffodil-core/src/main/scala/org/apache/daffodil/grammar/primitives/PrimitivesTextNumber.scala

```diff
@@ -63,36 +142,72 @@ case class ConvertTextNumberPrim(e: ElementBase)
     EntityReplacer { _.replaceForUnparse(zr) }
   }
 
-  val textNumberFormatEv: TextNumberFormatEv = {
-    val (pattern, patternStripped) = {
-      val p = e.textNumberPattern
-
-      if (p.startsWith(";")) {
-        e.SDE("Invalid textNumberPattern: The postive number pattern is mandatory")
-      }
+  private val pattern = {
+    val p = e.textNumberPattern
+    if (p.startsWith(";")) {
+      e.SDE("Invalid textNumberPattern: The postive number pattern is mandatory")
+    }
+    p
+  }
 
-      val noEscapedTicksRegex = """''""".r
-      val patternNoEscapedTicks = noEscapedTicksRegex.replaceAllIn(p, "")
-      val noQuotedRegex = """'[^']+'""".r
-      val patternNoQuoted = noQuotedRegex.replaceAllIn(patternNoEscapedTicks, "")
+  private val patternStripped = {
+    val noEscapedTicksRegex = """''""".r
+    val patternNoEscapedTicks = noEscapedTicksRegex.replaceAllIn(pattern, "")
+    val noQuotedRegex = """'[^']+'""".r
+    val patternNoQuoted = noQuotedRegex.replaceAllIn(patternNoEscapedTicks, "")
+    patternNoQuoted
+  }
 
-      if (patternNoQuoted.contains("V")) {
-        e.notYetImplemented("textNumberPattern with V symbol")
-      }
+  private val hasV = patternStripped.contains("V")
+  private val hasP = patternStripped.contains("P")
 
-      if (patternNoQuoted.contains("P")) {
+  private val textDecimalVirtualPoint = { // analogous to the property dfdl:binaryDecimalVirtualPoint
+    if (!hasV && !hasP) 0 // no virtual point
+    else {
+      if (hasP) {
         e.notYetImplemented("textNumberPattern with P symbol")
       }
+      Assert.invariant(hasV)
+      Assert.invariant(e.textNumberRep eq TextNumberRep.Standard)
+      //
+      // check for things incompatible with "V"
+      //
+      e.schemaDefinitionUnless(e.textStandardBase == 10,
+        "The dfdl:textNumberPattern V (virtual decimal point) requires that " +
+          "dfdl:textStandardBase is 10, but its value was %s",
+        e.textStandardBase)
+      //
+      val virtualPoint = TextNumberPatternUtils.textDecimalVirtualPointForStandard(e, patternStripped)
+
+      Assert.invariant(virtualPoint >= 1) // if this fails the regex is broken.
+      virtualPoint
+    }
+  }
+
+  private val patternWithoutPV =
+    pattern.replaceAll("P", "").replaceAll("V", "")
+
+  private val runtimePattern =
+    if (hasV || hasP) patternWithoutPV else pattern
+
+  val textNumberFormatEv: TextNumberFormatEv = {
 
       // Load the pattern to make sure it is valid
       try {
-        new DecimalFormat(p)
+        if (!hasV && !hasP)
+          new DecimalFormat(pattern) // But only if it doesn't have P nor V in it, since ICU doesn't support those
```

Comment:
Add comment about this check being unnecessary for P, V case since regex checking of pattern allows only a tiny subset of all that ICU's patterns can do, and we're certain this will be error free.